daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/IdeLedgerClient.scala

```diff
@@ -278,7 +278,9 @@ class IdeLedgerClient(
           }
         } else {
           // Allocate a fresh name based on the display name.
-          val candidates = displayName #:: LazyList.from(1).map(displayName + _.toString())
+          // Empty party ids are not allowed, fall back to "party" on empty display name.
+          val namePrefix = if (displayName.isEmpty) { "party" } else { displayName }
+          val candidates = namePrefix #:: LazyList.from(1).map(namePrefix + _.toString())
```

Comment:
I'm not really sure about this solution. Is this also what happens when allocating on a ledger via Script? Would it make sense to error out on the client with a more telling error?

Reply:
Running against sandbox with a script sandbox accepts this and assigns some automatic party identifier.

Reply:
Is it `"party"` or something else?

Reply:
It generates identifiers of the form `"party-0f47e587", "party-39f0402f"`, independent of the display-name. See [here for participant-state](https://github.com/digital-asset/daml/blob/4dd5663c5b38c4aaf4968325cb902c18bd387d84/ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/api/KeyValueParticipantStateWriter.scala#L96-L97) and [here for sandbox-classic](https://github.com/digital-asset/daml/blob/4dd5663c5b38c4aaf4968325cb902c18bd387d84/ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/ledger/PartyIdGenerator.scala#L12).
The script service is different in that it uses the display-name and appends a consecutive count if the name is already taken.

Reply:
Oh, I see, thanks.